### PR TITLE
[Fix] Fix the bug that when all pixels in an image is ignored, the ac…

### PR DESCRIPTION
**Pull request by RockeyCoss**
…curacy calculation raises ZeroDivisionError

Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Fix the bug that when all pixels in an image is ignored, the accuracy computation raises ZeroDivisionError

## Modification

Fix the bug that when all pixels in an image is ignored, the accuracy computation raises ZeroDivisionError

### CODE REVIEW

File: mmseg/models/losses/accuracy.py [hunk 1]
```diff
@@ -48,8 +48,13 @@ def accuracy(pred, target, topk=1, thresh=None, ignore_index=None):
...
         correct_k = correct[:k].reshape(-1).float().sum(0, keepdim=True)
-        res.append(
-            correct_k.mul_(100.0 / target[target != ignore_index].numel()))
+        total_num = target[target != ignore_index].numel()
+        # Avoid causing ZeroDivisionError when all pixels
+        # of an image are ignored
+        if total_num != 0:
+            res.append(correct_k.mul_(100.0 / total_num))
+        else:
+            res.append(100.0)
```
Comment (MeowZheng) [inline mmseg/models/losses/accuracy.py:57]: `100.` might not be proper.
just check the effect in `mAcc` and mmseg/models/decode_heads/point_head.py

**Review by MeowZheng — APPROVED** [review]

**Review by MeowZheng — COMMENTED** [review]
please fix ut error

**Review by MeowZheng — APPROVED** [review]

**Review by MengzhangLI — APPROVED** [review]

File: mmseg/models/losses/accuracy.py [hunk 2]
```diff
@@ -46,10 +46,13 @@ def accuracy(pred, target, topk=1, thresh=None, ignore_index=None):
         correct = correct & (pred_value > thresh).t()
     correct = correct[:, target != ignore_index]
     res = []
+    eps = 1e-6
```
Comment (Junjun2016) [inline mmseg/models/losses/accuracy.py:49]: Should we use eps in `torch.finfo`?
https://pytorch.org/docs/stable/type_info.html?highlight=finfo#torch.torch.finfo

Comment (Junjun2016) [reply mmseg/models/losses/accuracy.py:49]: @MeowZheng

**Review by Junjun2016 — APPROVED** [review]
LGTM, except for the definition of `eps`.

File: mmseg/models/losses/accuracy.py (hunk 2 above)
Comment (Junjun2016) [reply mmseg/models/losses/accuracy.py:49]: Please also refer to https://discuss.pytorch.org/t/torch-epsilon/30497.

**Pull request merged**